homeassistant/components/isy994/number.py

```diff
@@ -73,9 +105,102 @@ async def async_setup_entry(
             )
         )
 
+    for node, control in isy_data.aux_properties[Platform.NUMBER]:
+        control_detail = CONTROL_DESC[control]
+        name = COMMAND_FRIENDLY_NAME.get(control, control).replace("_", " ").title()
+        if node.address != node.primary_node:
+            name = f"{node.name} {name}"
+        description_on_level = NumberEntityDescription(
+            key=f"{isy_data.uid_base(node)}_{control}",
+            name=name,
+            native_unit_of_measurement=cast(
+                str, control_detail[ATTR_UNIT_OF_MEASUREMENT]
+            ),
+            entity_category=EntityCategory.CONFIG,
+            native_min_value=cast(float, control_detail[ATTR_MIN]),
+            native_max_value=cast(float, control_detail[ATTR_MAX]),
+            native_step=cast(float, control_detail[ATTR_STEP]),
+        )
+
+        entities.append(
+            ISYAuxControlNumberEntity(
+                node=node,
+                control=control,
+                unique_id=f"{isy_data.uid_base(node)}_{control}",
+                description=description_on_level,
+                device_info=device_info.get(node.primary_node),
+            )
+        )
     async_add_entities(entities)
 
 
+class ISYAuxControlNumberEntity(NumberEntity):
+    """Representation of a ISY/IoX Aux Control Number entity."""
+
+    _attr_mode = NumberMode.SLIDER
+    _attr_should_poll = False
+    _aux_control: NodeProperty | None
+
+    def __init__(
+        self,
+        node: Node,
+        control: str,
+        unique_id: str,
+        description: NumberEntityDescription,
+        device_info: DeviceInfo | None,
+    ) -> None:
+        """Initialize the ISY Aux Control Number entity."""
+        self._node = node
+        self._control = control
+        self.entity_description = description
+        self._attr_has_entity_name = node.address == node.primary_node
+        self._attr_unique_id = unique_id
+        self._attr_device_info = device_info
+        self._attr_native_value = None
+        self._change_handler: EventListener | None = None
+
+    async def async_added_to_hass(self) -> None:
+        """Subscribe to the node control change events."""
+        self._change_handler = self._node.control_events.subscribe(self.async_on_update)
+
+    @callback
+    def async_on_update(self, event: NodeProperty) -> None:
+        """Handle a control event from the ISY Node."""
+        if event.control != self._control:
+            return
+        self.async_write_ha_state()
```

Comment:
Ideally the reject filtering is a higher level up and this never gets called if `event.control != self._control` but that may not be viable right now

Reply:
Right now the module calls the subscriber for every event received for that node, no option to select the desired property to listen to. I'll add it to the list for future changes in PyISY

Reply:
It's also still on my road map to move the event logic out of the entities entirely (including isy994_control events).

Reply:
https://github.com/automicus/PyISY/issues/364